main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala

```diff
@@ -530,6 +530,28 @@ object ResolutionError {
     }
   }
 
+  /**
+    * An error raise to indicate a superclass cycle.
+    *
+    * @param path the superclass path from a class to itself.
+    * @param loc  the location where the error occurred.
+    */
+  case class SuperclassCycle(path: List[Symbol.ClassSym], loc: SourceLocation) extends ResolutionError {
+    override def summary: String = "Superclass cycle"
```

Comment:
We need to show the cycle in the summary. VSCode only shows the summary.

Reply:
And it has to be one line, right?

We could do `Superclass cycle: 'A' extends 'B' extends 'C' extends 'A'` ?

Reply:
That's right. You can thank Microsoft for that :)